apps/api/v2/src/ee/bookings/2024-08-13/controllers/bookings.controller.ts

```diff
@@ -0,0 +1,159 @@
+  async createBooking(
+    @Body(new CreateBookingInputPipe())
+    body: CreateBookingInput,
+    @Req() request: Request
+  ): Promise<CreateBookingOutput_2024_08_13> {
+    const booking = await this.bookingsService.createBooking(request, body);
+
+    if (Array.isArray(booking)) {
+      await this.bookingsService.billBookings(booking);
+    } else {
+      await this.bookingsService.billBooking(booking);
+    }
+
+    return {
+      status: SUCCESS_STATUS,
+      data: booking,
+    };
+  }
+
+  @Get("/:bookingUid")
+  async getBooking(@Param("bookingUid") bookingUid: string): Promise<GetBookingOutput_2024_08_13> {
+    if (!bookingUid) {
+      throw new BadRequestException("Booking UID is required in request path /:bookingUid");
+    }
+
+    const booking = await this.bookingsService.getBooking(bookingUid);
+
+    return {
+      status: SUCCESS_STATUS,
+      data: booking,
+    };
+  }
+
+  @Get("/")
+  @UseGuards(ApiAuthGuard)
+  @Permissions([BOOKING_READ])
+  async getBookings(
+    @Query() queryParams: GetBookingsInput_2024_08_13,
+    @GetUser() user: User
+  ): Promise<GetBookingsOutput_2024_08_13> {
+    const bookings = await this.bookingsService.getBookings(queryParams, user);
+
+    return {
+      status: SUCCESS_STATUS,
+      data: bookings,
+    };
+  }
+
+  @Post("/:bookingUid/reschedule")
+  async rescheduleBooking(
+    @Param("bookingUid") bookingUid: string,
+    @Body() body: RescheduleBookingInput_2024_08_13,
+    @Req() request: Request
+  ): Promise<RescheduleBookingOutput_2024_08_13> {
+    if (!bookingUid) {
+      throw new BadRequestException("Booking UID is required in request path /:bookingUid/reschedule");
+    }
+
+    const newBooking = await this.bookingsService.rescheduleBooking(request, bookingUid, body);
+    await this.bookingsService.billRescheduledBooking(newBooking, bookingUid);
+
+    return {
+      status: SUCCESS_STATUS,
+      data: newBooking,
+    };
+  }
+
+  @Post("/:bookingUid/cancel")
+  @HttpCode(HttpStatus.OK)
+  async cancelBooking(
+    @Req() request: Request,
+    @Param("bookingUid") bookingUid: string,
+    @Body() body: CancelBookingInput_2024_08_13
+  ): Promise<CancelBookingOutput_2024_08_13> {
+    if (!bookingUid) {
+      throw new BadRequestException("Booking UID is required in request path /:bookingUid/cancel");
+    }
+
+    const cancelledBooking = await this.bookingsService.cancelBooking(request, bookingUid, body);
+
+    return {
+      status: SUCCESS_STATUS,
+      data: cancelledBooking,
+    };
+  }
+
+  @Post("/:bookingUid/mark-absent")
+  @HttpCode(HttpStatus.OK)
+  @Permissions([BOOKING_WRITE])
+  @UseGuards(ApiAuthGuard)
+  async markNoShow(
+    @Param("bookingUid") bookingUid: string,
+    @Body() body: MarkAbsentBookingInput_2024_08_13,
+    @GetUser("id") ownerId: number
+  ): Promise<MarkAbsentBookingOutput_2024_08_13> {
+    if (!bookingUid) {
+      throw new BadRequestException("Booking UID is required in request path /:bookingUid/cancel");
```

Comment:
```suggestion
      throw new BadRequestException("Booking UID is required in request path /:bookingUid/mark-absent");
```

Reply:
Fixed by adding guard that throws bad request exception in  [feat: BookingUidGuard](https://github.com/calcom/cal.com/pull/16200/commits/45b593bb27210e5f32367d3077112498c9340690) so this error message was removed all together.